src/test/java/rx/subscriptions/CompositeSubscriptionTest.java

```diff
@@ -344,4 +344,121 @@ public void testAddingNullSubscriptionIllegal() {
         csub.add(null);
     }
 
+    @Test
+    public void testAddAll() {
+        final AtomicInteger counter = new AtomicInteger();
+        CompositeSubscription s = new CompositeSubscription();
+        s.addAll(new Subscription() {
+            @Override
+            public void unsubscribe() {
+                counter.incrementAndGet();
+            }
+
+            @Override
+            public boolean isUnsubscribed() {
+                return false;
+            }
+        }, new Subscription() {
+            @Override
+            public void unsubscribe() {
+                counter.incrementAndGet();
+            }
+
+            @Override
+            public boolean isUnsubscribed() {
+                return false;
+            }
+        }, new Subscription() {
+            @Override
+            public void unsubscribe() {
+                counter.incrementAndGet();
+            }
+
+            @Override
+            public boolean isUnsubscribed() {
+                return false;
+            }
+        }, new Subscription() {
+            @Override
+            public void unsubscribe() {
+                counter.incrementAndGet();
+            }
+
+            @Override
+            public boolean isUnsubscribed() {
+                return false;
+            }
+        });
+
+        s.unsubscribe();
+
+        assertEquals(4, counter.get());
+    }
+
+    @Test(timeout = 1000)
+    public void testAddAllConcurrent() throws InterruptedException {
```

Comment:
Not sure about this test since it gives no exact guarantees about thread-safety of the target method, depends on the test environment.

Also, current timeout may give flakiness on Travis (sometimes it has significant delays in threads scheduling)